feg/gateway/services/session_proxy/credit_control/gx/gx_client.go

```diff
@@ -244,8 +245,18 @@ func (gxClient *GxClient) createCreditControlMessage(
 	}
 
 	apn := datatype.UTF8String(request.Apn)
-	if globalConfig != nil && len(globalConfig.PCFROverwriteApn) > 0 {
-		apn = datatype.UTF8String(globalConfig.PCFROverwriteApn)
+	if globalConfig != nil {
+		if len(globalConfig.VirtualApnRules) > 0 {
+			for _, rule := range globalConfig.VirtualApnRules {
```

Comment:
Since this logic is the same for both Gx and Gy, I would pull this logic into a function tied to `VirtualApnRule` and use it here. Once the other overwrite APN configs (PCRF/OCS) are deprecated this should look cleaner.

Reply:
Not sure if we gain anything from factoring these three lines but anyway, I added this.